setup/src/Magento/Setup/Module/Di/Compiler/Config/Reader.php

```diff
@@ -100,11 +100,6 @@ public function generateCachePerScope(
             $config['instanceTypes'][$virtualType] = $areaConfig->getInstanceType($virtualType);
         }
 
-        // sort configuration to have it in the same order on every build
-        ksort($config['arguments']);
-        ksort($config['preferences']);
-        ksort($config['instanceTypes']);
```

Comment:
Note: this code was moved to `setup/src/Magento/Setup/Module/Di/App/Task/Operation/Area.php`